web_shortcuts/model/web_shortcut.py

```diff
@@ -39,14 +39,17 @@ class web_shortcut(models.Model):
     @api.model
     def get_user_shortcuts(self, user_id):
         shortcuts = self.search([('user_id', '=', user_id)])
-        results = shortcuts.read(['menu_id'])
-        ir_ui_menu_obj = self.env['ir.ui.menu']
-        menus = ir_ui_menu_obj.search([('id', 'in', [x['menu_id'][0]
-                                                     for x in results])])
-        name_map = dict(menus.name_get())
-        # Make sure to return only shortcuts pointing to existing menu items.
-        filtered_results = filter(lambda result: result['menu_id'][0] in
-                                  name_map, results)
-        for result in filtered_results:
-            result.update(name=name_map[result['menu_id'][0]])
-        return filtered_results
+        res = []
+        for shortcut in shortcuts:
```

Comment:
I tried to stay compatible with the old code :
```python
filtered_results = [{
'menu_id': (543, u'Messaging/Users/Users'), 
'id': 26,  <--- come from the shortcut object
'name':  u'Messaging/Users/Users'
}]
```
If you agree we can manipulate ir.ui.menu instead of shortcuts
```xml
<t t-name="Systray.Shortcuts.Item">
    <li>
        <a href="#" t-att-data-id="shortcut.menu_id[0]" t-att-data-shortcut-id="shortcut.id">
            <t t-esc="shortcut.name"/>
        </a>
    </li>
</t>
```